static/panes/ast-view.js

```diff
@@ -157,6 +171,29 @@ Ast.prototype.onCompiler = function (id, compiler, options, editorid) {
     }
 };
 
+Ast.prototype.onColours = function (id, colours, scheme) {
+    this.lastColours = colours;
```

Comment:
I think you should probably not latch these unless `id === this._compilerId`, otherwise a second compilation window could confuse things?

Reply:
From what's up I imagine this is also copy/paste code? In which case we can raise an issue to look at this (and the source) separately.

Reply:
Makes sense. So far, the confusion does not manifest itself, because these values used only in `Ast.prototype.onCompileResult` there they are only read if the recompilation completed with the appropriate compiler.
I'll add the condition.

Reply:
> From what's up I imagine this is also copy/paste code? In which case we can raise an issue to look at this (and the source) separately.
Yep. Here is the source:
https://github.com/compiler-explorer/compiler-explorer/blob/main/static/panes/ir-view.js#L194